### PR TITLE
Posit.RStudio.OpenSource 2022.12.0+353

### PR DESCRIPTION
**Pull request by dpprdan**
[publisher change](https://github.com/microsoft/winget-pkgs/blob/master/FAQ.md?rgh-link-date=2023-03-22T15%3A38%3A17Z#what-should-i-do-if-a-package-is-being-published-by-a-new-publisher): RStudio -> [Posit Software](https://posit.co/), cf  #100319

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/100321)

### CODE REVIEW

**Comment by dpprdan** [comment]
@cderv This is the PR to make the publisher change from RStudio to Posit, [as mentioned earlier](https://github.com/microsoft/winget-pkgs/pull/91788#issuecomment-1358435593).

**Comment by cderv** [comment]
It looks ok to me - you know the rules and guidelines to change published better. 

Thanks for keeping this up to date !

**Comment by stephengillie** [comment]
Hi @dpprdan,

The package installs normally, but requires the R programming language to run:
image

I couldn't find the R language among the WinGet public manifests. It might need to be added before it can be listed as a dependency. 

Pending:
* https://github.com/microsoft/winget-cli/issues/163

**Comment by cderv** [comment]
> I couldn't find the R language among the WinGet public manifests.

FWIW it is called Rproject.R

````
> winget search Rproject.R
Nom           ID              Version Source
---------------------------------------------
R for Windows RProject.R      4.2.3   winget
Rtools        RProject.Rtools 4.2     winget
````

> This PR is blocked until support for dependencies is implemented in:

Is this new that IDE tools can't be added in winget anymore until support for dependency is added ? 
This tool is available already but with another publisher. Company has changed name so it is republished with new name. 

````
> winget search RStudio.OpenSource
Nom                                 ID                         Version       Source
------------------------------------------------------------------------------------
RStudio Desktop Open Source Edition RStudio.RStudio.OpenSource 2022.12.0+353 winget
````

I don't expect installing an IDE with winget to install language tools with it.

**Comment by dpprdan** [comment]
@stephengillie Like @cderv wrote (and I did in the original post), this is just a publisher change
I.e. RStudio.RStudio.OpenSource has been on winget-pkgs for quite a while (in fact [you approved](https://github.com/microsoft/winget-pkgs/pull/100319) the lastest changes to the manifest yourself) and this just changes it to Posit.RStudio.OpenSource. 

So I don't understand, why needing a dependency should matter now.

File: manifests/p/Posit/RStudio/OpenSource/2022.12.0+353/Posit.RStudio.OpenSource.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+- interactive
+- silent
+- silentWithProgress
+FileExtensions:
+- r
+- rmd
+AppsAndFeaturesEntries:
+  - Publisher: Posit Software
+    DisplayName: RStudio
+    ProductCode: RStudio
```
Comment (stephengillie) [inline manifests/p/Posit/RStudio/OpenSource/2022.12.0+353/Posit.RStudio.OpenSource.installer.yaml:16]:
```suggestion
    ProductCode: RStudio
Dependencies:
  PackageDependencies:
    - PackageIdentifier: RProject.R
```

**Comment by stephengillie** [comment]
> > I couldn't find the R language among the WinGet public manifests.
> 
> FWIW it is called Rproject.R
> 
> ```
> > winget search Rproject.R
> Nom           ID              Version Source
> ---------------------------------------------
> R for Windows RProject.R      4.2.3   winget
> Rtools        RProject.Rtools 4.2     winget
> ```

Thank you. With this package preinstalled, the PR's package `Posit.RStudio` installs and runs normally, both CLI and GUI. 

> > This PR is blocked until support for dependencies is implemented in:
> 
> Is this new that IDE tools can't be added in winget anymore until support for dependency is added ? This tool is available already but with another publisher. Company has changed name so it is republished with new name.
> 
> ```
> > winget search RStudio.OpenSource
> Nom                                 ID                         Version       Source
> ------------------------------------------------------------------------------------
> RStudio Desktop Open Source Edition RStudio.RStudio.OpenSource 2022.12.0+353 winget
> ```
> 
> I don't expect installing an IDE with winget to install language tools with it.

Apologies for blocking an existing package. If you'll accept adding the dependency suggested above, I'll manually validate this package as an upgrade to an existing package with a dependency. Adding the dependency will ensure that after https://github.com/microsoft/winget-cli/issues/163 new users will also get the language dependency.

**Comment by dpprdan** [comment]
@stephengillie Thanks, done!

**Review by ItzLevvie — APPROVED** [review]

**Comment by cderv** [comment]
> Adding the dependency will ensure that after https://github.com/microsoft/winget-cli/issues/163 new users will also get the language dependency.

@stephengillie does this mean that installing this tool will also install R using winget ? Or will it just warn user about a needed dependency ?  I guess this is what https://github.com/microsoft/winget-cli/issues/163 is about ?

**Comment by stephengillie** [comment]
> @stephengillie does this mean that installing this tool will also install R using winget ? Or will it just warn user about a needed dependency ? I guess this is what [microsoft/winget-cli#163](https://github.com/microsoft/winget-cli/issues/163) is about ?

Yes, users installing this tool will also have R installed, since R is needed to use this tool. But if they already have R installed, then it won't be installed again. The goal is for installing this tool ensures that R is also on the system, so it may be used. This is the goal of cli#163.

**Pull request merged**